tests/tensor/test_nlinalg.py

```diff
@@ -664,8 +685,10 @@ def test_eval(self):
         tf_b1 = function([B], [Bi1])
         t_binv = tf_b(self.b)
         t_binv1 = tf_b1(self.b1)
-        assert _allclose(t_binv, n_binv)
-        assert _allclose(t_binv1, n_binv1)
+        atol_, rtol_ = _get_atol_rtol(np.asarray(t_binv), n_binv)
+        assert np.allclose(t_binv, n_binv, atol=atol_, rtol=rtol_)
+        atol_, rtol_ = _get_atol_rtol(np.asarray(t_binv1), n_binv1)
+        assert np.allclose(t_binv1, n_binv1, atol=atol_, rtol=rtol_)
```

Comment:
This is too much code to recreate every time it's used. The question is whether we really need the inner logic. Seems like it has some special way of deciding tolerances for float16/32 complex64. If the tests are covering these types we can hardcode the tolerances, the same way we usually do `atol = x if config.floatX else y`. But I suspect many of these tests don't even make use of these specific types (I could be wrong).

Otherwise we should keep the original function and not repeat the code so many times in the tests. These functions that are only used in tests should be moved to the tests package, out of the main pytensor package (if we end up keeping it).

Also always use `np.testing.assert_allclose` and not `np.allclose`, as it has a nicer output when it fails.